### PR TITLE
update virtual list to support on screen change

### PR DESCRIPTION
**Pull request by lneir**
adds support for callback 'onScreenChanged' that will notify when an item has come onto the screen or left the screen view.

### CODE REVIEW

File: extensions/virtuallistview/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "reactxp-virtuallistview",
-  "version": "0.1.9",
+  "version": "0.1.9-sym-1.0",
```
Comment (erictraut) [inline extensions/virtuallistview/package.json]: Please don't change the package version. I'll increment it if/when I publish a new official version.

File: extensions/virtuallistview/src/VirtualListView.tsx [hunk 2]
```diff
@@ -1010,8 +1010,11 @@ export class VirtualListView<ItemInfo extends VirtualListViewItemInfo>
         // Set the "live" values as well.
         let cell = this.refs[cellInfo.virtualKey] as VirtualListCell;
         if (cell) {
+            // on screen when full message is within layoutHeight area.
+            let isOnScreen = top > this._lastScrollTop && (top + cellInfo.height) < this._lastScrollTop + this._layoutHeight;
```
Comment (erictraut) [inline extensions/virtuallistview/src/VirtualListView.tsx]: This check requires that the item be on screen in its entirety, as opposed to partly on screen. It's not clear to me that's what all callers will want. It depends on the use case.

File: extensions/virtuallistview/src/VirtualListCell.tsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ import RX = require('reactxp');
 
 export interface VirtualListCellInfo {
     key: string;
+    onScreenChanged?: (isOnScreen: boolean) => void;
```
Comment (erictraut) [inline extensions/virtuallistview/src/VirtualListCell.tsx]: An interface change like this should be accompanied by an update to the documentation (VirtualListView.md).

Comment (erictraut) [inline extensions/virtuallistview/src/VirtualListCell.tsx]: The terminology here isn't ideal. "Screen" usually implies the full physical display surface, whereas this is referring to the view port of the list.

File: extensions/virtuallistview/src/VirtualListCell.tsx [hunk 4]
```diff
@@ -223,6 +225,13 @@ export class VirtualListCell extends RX.Component<VirtualListCellProps, null> {
         return this._isVisible;
     }
 
+    setIsOnScreen(isOnScreen: boolean) {
```
Comment (erictraut) [inline extensions/virtuallistview/src/VirtualListCell.tsx]: I don't like exposing the notion of "visible in view port" at the list cell level. A list cell shouldn't have this knowledge. The visible range is something that's maintained by the list, so if we add a callback, it seems like it should be exposed by the VirtualListView, not the VirtualListCell.

File: extensions/virtuallistview/package.json (hunk 1 above)
Comment (govindvarshney) [reply extensions/virtuallistview/package.json]: This was more done for internal purposes. We have our own internal repository.

**Comment by govindvarshney** [comment]
"_We've talked about maybe adding the same interface to VirtualListView, but we haven't had a use case for it so far. Would this solution meet your needs?_" That would be ideal.

**Pull request closed without merging**